Lib/typing.py

```diff
@@ -1444,54 +1454,68 @@ def _determine_new_args(self, args):
 
         new_args = []
         for old_arg in self.__args__:
+            self._make_substitution(old_arg, new_args, new_arg_by_param)
+        return tuple(new_args)
 
-            if isinstance(old_arg, type):
-                new_args.append(old_arg)
-                continue
+    def _make_substitution(self, old_arg, new_args, new_arg_by_param):
+        # Mutates `new_args` and inserts the correct new argument.
```

Comment:
I'd find it this function much more readable if it returned a new list rather than mutating the list that's passed in :/

```suggestion
        """Mutate `new_args` and insert the correct new argument."""
```

Reply:
I used the mutation approach, because it required less code changes.
But, returning a new one also works.

Reply:
Refactored.